include/alpaka/mem/buf/oacc/Copy.hpp

```diff
@@ -335,29 +335,40 @@ namespace alpaka
             {
                 ALPAKA_DEBUG_FULL_LOG_SCOPE;
 
-                ALPAKA_ASSERT(
-                    getDev(viewDst).m_spDevOaccImpl->iDevice() == getDev(viewSrc).m_spDevOaccImpl->iDevice());
-
-                return alpaka::oacc::detail::
-                    makeTaskCopyOacc<alpaka::oacc::detail::TaskCopyOacc, TDim, TViewDst, TViewSrc, TExtent>(
-                        viewDst,
-                        viewSrc,
-                        extent,
-                        getDev(viewDst),
 #    if _OPENACC >= 201510 && (!defined __GNUC__)
-                        acc_memcpy_device
-#    else
-                        // acc_memcpy_device is only available since OpenACC2.5
-                        // , but we want the tests to compile anyway
-                        [](void* dst, void* src, std::size_t size)
-                        {
-                            void* buf = core::alignedAlloc(core::vectorization::defaultAlignment, size);
-                            acc_memcpy_from_device(buf, src, size);
-                            acc_memcpy_to_device(dst, buf, size);
-                            core::alignedFree(buf);
-                        }
+                // acc_memcpy_device is only available since OpenACC2.5, but we want the tests to compile anyway
+                if(getDev(viewDst).m_spDevOaccImpl->iDevice() == getDev(viewSrc).m_spDevOaccImpl->iDevice())
+                {
+                    return alpaka::oacc::detail::
+                        makeTaskCopyOacc<alpaka::oacc::detail::TaskCopyOacc, TDim, TViewDst, TViewSrc, TExtent>(
+                            viewDst,
+                            viewSrc,
+                            extent,
+                            getDev(viewDst),
+                            acc_memcpy_device);
+                }
+                else
 #    endif
-                    );
+                {
+                    return alpaka::oacc::detail::
+                        makeTaskCopyOacc<alpaka::oacc::detail::TaskCopyOacc, TDim, TViewDst, TViewSrc, TExtent>(
+                            viewDst,
+                            viewSrc,
+                            extent,
+                            getDev(viewDst),
+                            [devSrc = getDev(viewSrc),
+                             devDst = getDev(viewDst)](void* dst, void* src, std::size_t size)
+                            {
+                                static const auto alignedDeleter = [](void* p) { core::alignedFree(p); };
+                                std::unique_ptr<void, decltype(alignedDeleter)> buf(
+                                    core::alignedAlloc(core::vectorization::defaultAlignment, size),
+                                    alignedDeleter);
```

Comment:
Can you not just pass `core::alignedFree` directly?

```suggestion
                                std::unique_ptr<void, decltype(&core::alignedFree)> buf(
                                    core::alignedAlloc(core::vectorization::defaultAlignment, size),
                                    &core::alignedFree);
```

Reply:
Sure. I  messed up using `decltype` to get function *pointer* type m( .